File: 1013-partition-array-into-three-parts-with-equal-sum/1013-partition-array-into-three-parts-with-equal-sum.py
class Solution(object):
    def canThreePartsEqualSum(self, arr):
        total = sum(arr)
        curr = 0
        count = 0
        if(total % 3 != 0):
            return False
        actual = total//3
        for i in range(len(arr)):
            curr += arr[i]
            if(curr == actual):
                curr = 0
                count+=1    
            if count == 2 and i <  len(arr) - 1:
                return True
        return False     

File: 1013-partition-array-into-three-parts-with-equal-sum/1013-partition-array-into-three-parts-with-equal-sum.cpp
class Solution {
public:
    bool canThreePartsEqualSum(vector<int>& arr) {
        int count = 0;
        int sum = 0;
        for (int c : arr){
           sum += c;
        }
        int total = sum / 3;
        int n = arr.size();
        int cur = 0;

        if(sum % 3 != 0)
        return false;

        for(int i = 0; i < n; i++){
            cur += arr[i];
            if (cur == total){
                cur = 0;    
                count++;
            }
            if(count == 2 && i < arr.size() -1){
            return true;
            }
        }
        
        return false;
    }
};